src/Npgsql/NpgsqlConnector.Auth.cs

```diff
@@ -366,7 +366,7 @@ public override long Position
 
         class AuthenticationCompleteException : Exception { }
 
-        string? GetPassword()
+        string? GetPassword(string username)
```

Comment:
Can you please pass `username` to PgPassFile.Load as well, just like you do with ProvidePasswordCallback? The issue was there before you but we may as well fix it :)

Same for `database` (see comment below).